cli/cluster/cmd/publish.go

```diff
@@ -109,10 +116,45 @@ func replaceInstanceConfig(instance string, config *libcluster.Config,
 	}
 
 	cconfig, err = libcluster.ReplaceInstanceConfig(cconfig, instance, config)
-	if err != nil {
+	if err == nil {
+		// Instance is present in the configuration.
+		return libcluster.NewYamlConfigPublisher(publisher).Publish(cconfig.RawConfig)
+	}
```

Comment:
But what if the group and replicaset is wrong? I think a user should got an error in the case.

Reply:
The errors for this cases are added, and tests. Thank you!